uni/lib/controller/local_storage/app_shared_preferences.dart

```diff
@@ -176,8 +176,17 @@ class AppSharedPreferences {
   /// Returns a list containing the user's favorite widgets.
   static Future<List<FavoriteWidgetType>> getFavoriteCards() async {
     final prefs = await SharedPreferences.getInstance();
-    final storedFavorites = prefs.getStringList(favoriteCards);
-    if (storedFavorites == null) return defaultFavoriteCards;
+    final storedFavorites = prefs
+        .getStringList(favoriteCards)
+        ?.where(
+          (element) => int.parse(element) < FavoriteWidgetType.values.length,
+        )
```

Comment:
Is there any case that could cause this? In my mind, it's maybe acceptable if the number of cards are reducing from an update, and user_preferences is still storing the old favorites. But it doesn't solve the migration issue, because if we remove an card from the enum  eg.: `print_balance` which has an id of `2` and the user has an account, library or busStop cards, it might cause the card with highest id to "disappear" because now it will be filtered while it's still exists, or it might cause to appear a different card that the user didn't have before.

Reply:
You're right, this is a quick workaround to the case where a stored card no longer exists. (I got it by switching from the restaurants card branch). This will stop the app from crashing, but some weird things might happen since we are using indexes (e.g. wrong cards being added). I believe that we should think of a different way to store the cards, feel free to open an issue.

Reply:
Ok it seems okay to accept it in this state, but I'll open another issue to deal with migrations in cards